hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestLazyPartitionPathFetching.scala

```diff
@@ -55,6 +55,36 @@ class TestLazyPartitionPathFetching extends HoodieSparkSqlTestBase {
     }
   }
 
+  test("Test querying with date column + partition pruning") {
```

Comment:
How do you ensure partition pruning kicked in? Can you print the physical plan and paste here? If we can assert partition filters in the plan then it's even better. This is to safeguard any future change does not break the pruning.

Reply:
Like I've said before, we can't see any difference in the physical plan since all partition filters are pushed to HUDI side, just some filters can't take effect before. In order to make sure partition pruning take effect, I added `testPartitionPruneWithMultiplePartitionColumnsWithComplexExpression` and check `fileIndex.areAllPartitionPathsCached`, before this pr, the complex expression cannot be pushed, so `fileIndex.areAllPartitionPathsCached` return true, after this, it should return false.